### PR TITLE
Fix missing `HandOverDone` handler in ClusterSingletonManager

### LINKED ISSUE #6973: Akka.Cluster.Sharding: duplicate shards / entities
**Version Information**
Version of Akka.NET? v1.5 - all versions including v1.5.13
Which Akka.NET Modules? Akka.Cluster.Sharding

**Describe the bug**

We have very rough and loose data on this right now, but it's been reported by multiple users including on https://github.com/petabridge/Akka.Persistence.Azure/issues/350 - looks like there could be something wrong with Akka.Cluster.Sharding in v1.5 that allows a `Shard` to be allocated more than once. This is our thread to investigate.

## If you have run into this issue, provide a full dump of your config here please

### PR DESCRIPTION
**Pull request by Arkatufus**
Fixes https://github.com/akkadotnet/akka.net/issues/6973#issuecomment-2147626368

## Changes

Add `HandOverDone` message handler to ClusterSingletonManager Oldest state

### CODE REVIEW

File: src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManager.cs [hunk 1]
```diff
@@ -1179,60 +1179,58 @@ private void InitializeFSM()
...
-                }
-                else if (e.FsmEvent is SelfExiting)
-                {
-                    SelfMemberExited();
-                    // complete _memberExitingProgress when HandOverDone
-                    Sender.Tell(Done.Instance); // reply to ask
-                    return Stay();
-                }
-                else if (e.FsmEvent is MemberDowned memberDowned && e.StateData is OldestData od && memberDowned.Member.UniqueAddress.Equals(_cluster.SelfUniqueAddress))
+                switch (e.FsmEvent)
```
Comment (Arkatufus) [inline src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManager.cs:1182]: Convert if..else structure to switch..case (auto-refactor, no code change)

File: src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManager.cs [hunk 2]
```diff
@@ -1179,60 +1179,58 @@ private void InitializeFSM()
...
+                        }
+
                         return TryAcquireLease();
-                    }
+                    
+                    case HandOverDone:
+                        // no-op, the HandOverDone message can be sent multiple times if HandOverToMe
+                        // was sent multiple times (retried)
+                        // https://github.com/akka/akka/pull/29216/files#r440062592
+                        return Stay();
```
Comment (Arkatufus) [inline src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManager.cs:1229]: Actual fix, do a no-op when extra `HandOverDone` is received

**Review by Arkatufus — COMMENTED** [review]
Self review

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**